azurelinuxagent/ga/exthandlers.py

```diff
@@ -476,20 +479,45 @@ def handle_enable(self, ext_handler_i):
             if old_ext_handler_i is None:
                 ext_handler_i.install()
             elif ext_handler_i.version_ne(old_ext_handler_i):
-                old_ext_handler_i.disable()
-                ext_handler_i.copy_status_files(old_ext_handler_i)
-                if ext_handler_i.version_gt(old_ext_handler_i):
-                    ext_handler_i.update()
-                else:
-                    old_ext_handler_i.update(version=ext_handler_i.ext_handler.properties.version)
-                old_ext_handler_i.uninstall()
-                old_ext_handler_i.remove_ext_handler()
-                ext_handler_i.update_with_install()
+                ExtHandlersHandler._update_extension_handler(old_ext_handler_i, ext_handler_i)
         else:
             ext_handler_i.update_settings()
 
         ext_handler_i.enable()
 
+    @staticmethod
+    def _update_extension_handler(old_ext_handler_i, ext_handler_i):
+
+        def execute_old_handler_command(ext_handler_i, op):
+            """
+            Created a common function to execute all commands that need to be executed from the old handler
+            so that it can have a common exception handling mechanism
+            :param ext_handler_i: The extension handler instance to run the operation on
+            :param op: Which operation to run
+            """
+            try:
+                if op == WALAEventOperation.Disable:
```

Comment:
suggestion: if you pass a lambda then you don't need the switch on operation code

Reply:
Nice! Didnt know about it, will implement it!